.azure-pipelines/templates/stages/deploy-stage.yml

```diff
@@ -53,10 +54,16 @@ stages:
             inputs:
               secureFile: snapcraft.cfg
           - bash: |
+              set -e
               mkdir -p .snapcraft
               ln -s $(snapcraftCfg.secureFilePath) .snapcraft/snapcraft.cfg
               for SNAP_FILE in snap/*.snap; do
-                snapcraft upload --release=${{ parameters.snapReleaseChannel }} "${SNAP_FILE}"
+                for i in `seq 1 3`;
+                do
+                  if snapcraft upload --release=${{ parameters.snapReleaseChannel }} "${SNAP_FILE}" ; then
+                    break
+                  fi
+                done
```

Comment:
It seems like the error is still hidden if `snap upload` fails three times. I put the following in a file:
```
#!/bin/bash
set -e
for i in `seq 1 3`; 
do
  if cat nonexistent; then
    break
  fi  
done
```
and see this when running it:
```
% ./test.sh 
cat: nonexistent: No such file or directory
cat: nonexistent: No such file or directory
cat: nonexistent: No such file or directory
% echo $?
0
```
I think we should fix this so we know if the upload fails. What do you think?